bot/exts/utils/snekbox.py

```diff
@@ -116,16 +117,75 @@ async def convert(cls, ctx: Context, code: str) -> list[str]:
         return codeblocks
 
 
+class PythonVersionSwitcherButton(ui.Button):
+    """A button that allows users to re-run their eval command in a different Python version."""
+
+    def __init__(
+        self,
+        job_name: str,
+        version_to_switch_to: Literal["3.10", "3.11"],
+        snekbox_cog: "Snekbox",
+        ctx: Context,
+        code: str
+    ) -> None:
+        self.version_to_switch_to = version_to_switch_to
+        super().__init__(label=f"Run in {self.version_to_switch_to}", style=enums.ButtonStyle.primary)
+
+        self.snekbox_cog = snekbox_cog
+        self.ctx = ctx
+        self.job_name = job_name
+        self.code = code
+
+    async def callback(self, interaction: Interaction) -> None:
+        """
+        Tell snekbox to re-run the user's code in the alternative Python version.
+
+        Use a task calling snekbox, as run_job is blocking while it waits for edit/reaction on the message.
+        """
+        scheduling.create_task(self.snekbox_cog.run_job(self.job_name, self.ctx, self.version_to_switch_to, self.code))
+        await interaction.message.delete()
+
+
 class Snekbox(Cog):
     """Safe evaluation of Python code using Snekbox."""
 
     def __init__(self, bot: Bot):
         self.bot = bot
         self.jobs = {}
 
-    async def post_job(self, code: str, *, args: Optional[list[str]] = None) -> dict:
+    def build_python_version_switcher_view(
+        self,
+        job_name: str,
+        member: Member,
```

Comment:
This can be retrieved from the `ctx` argument and I don't think the flexibility of passing an arbitrary member is necessary.

Reply:
https://github.com/python-discord/bot/pull/2218/commits/47c60a12aa1ce9ad331bf56a84ffde318cb38a11